pilot/proxy/envoy/policy.go

```diff
@@ -23,6 +23,22 @@ import (
 	"istio.io/istio/pilot/proxy"
 )
 
+func isDestinationBlacklistedForAuth(serviceName string, mesh *proxyconfig.MeshConfig) bool {
+	hostname, _, _ := model.ParseServiceKey(serviceName)
+	for _, blacklisted_svc := range mesh.AuthBlacklistedDestinationService {
+		if hostname == blacklisted_svc {
+			return true
+		}
+	}
+	namespace := model.ExtractNamespaceFromHostname(hostname)
+	for _, blacklisted_namespace := range mesh.AuthBlacklistedDestinationNamespace {
+		if namespace == blacklisted_namespace {
+			return true
```

Comment:
this is not platform independent. We have no concept of namespace inside proxy/envoy. Thats the reason we use hostnames everywhere.

Reply:
Would it be ok if we drop blacklisted namespace? It's just for convenience to setup the blacklisted. I guess asking operators to list all services explicitly is not too much.

Reply:
See istio/api for a comment. we can drop the namespace and use hostnames with wildcards like *.myns.svc.cluster.local , such that when you are in this region of code, you do a regex match against service hostname and decide whether or not you want to black list this service. It keeps code platform independent.

Reply:
Done. Change to single list of fqdns for now. (I don't want to use regex yet, as it might cause some unexpected outcome, especially with all the dot in the service name).